### PR TITLE
Update base AMI to Debian Bookworm

### LINKED ISSUE #300: DNS woes with Debian Bookworm
# 🐛 Summary #

@mcdonnnj has reported DNS woes when using a Debian Bookworm base AMI.  He says that DNS resolution requires the FQDN (e.g., `kali1.envX.cool.cyber.dhs.gov`); simply using the hostname (e.g., `kali1`) does not work.

The reason for the difference appears to be that Bookworm uses `systemd-resolved` whereas previous Debian releases used `resolvconf`.

## To reproduce ##

1. Build a new COOL AMI from an AMI based on Debian Bookworm.
2. Deploy the new COOL AMI along with an old, Bullseye-based COOL AMI into a COOL environment.
3. Observe that, in the instance based on the new COOL AMI, DNS fails to resolve without using the FQDN; but, DNS _does_ resolve in the instance based on the old AMI without the FQDN.

## Expected behavior ##

The DNS behavior should remain the same, regardless of whether `systemd-resolved` or `resolvconf` is used.

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request updates the base AMI from Debian Bullseye to Debian Bookworm.

## 💭 Motivation and context ##

Debian Bookworm was officially released on June 10, 2023.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

**Review by mcdonnnj — APPROVED** [review]
image

**Review by dv4harr10 — APPROVED** [review]

**Review by jasonodoom — APPROVED** [review]

**Review by dav3r — APPROVED** [review]

**Comment by mcdonnnj** [comment]
@jsf9k Until we come to a final determination on handling the change in Bookworm from using [resolvconf](https://packages.debian.org/bookworm/resolvconf) to [systemd-resolved](https://packages.debian.org/bookworm/systemd-resolved) for DNS resolution would it be fair to mark this as blocked?

**Comment by jsf9k** [comment]
Please note that the linter failures are because this PR is lacking the changes from PR #295.  Once that PR is merged then this one should also pass its checks.

**Review by dav3r — APPROVED** [review]
👍 👍

File: src/fix-netplan-configuration.yml
Comment (mcdonnnj) [inline src/fix-netplan-configuration.yml:1]: Thoughts on changing the name of this playbook? It reads like it fixes the Netplan configuration when it's installing prerequisites for the downstream bits that fix the configuration.

Comment (dav3r) [reply src/fix-netplan-configuration.yml:1]: That is a good point - it confused me at first as well.  How about `install-prereqs-for-netplan-config-fix.yml`?  Do any of you have something more concise?

Comment (jsf9k) [reply src/fix-netplan-configuration.yml:1]: Please enjoy commit f0f49a1.

**Comment by jsf9k** [comment]
@mcdonnnj - Can you post something here and/or in #300 when your CyHy testing is complete?

**Comment by jsf9k** [comment]
> @mcdonnnj - Can you post something here and/or in #300 when your CyHy testing is complete?

@mcdonnnj - Does cisagov/cyhy_amis#746 mean that your testing is complete and successful?

**Comment by mcdonnnj** [comment]
> > @mcdonnnj - Can you post something here and/or in #300 when your CyHy testing is complete?
> 
> @mcdonnnj - Does [cisagov/cyhy_amis#746](https://github.com/cisagov/cyhy_amis/pull/746) mean that your testing is complete and successful?

It does now. I meant to create that as a draft PR while I did some last verification. I have finished that verification and added a small update to #300 confirming success.

**Pull request merged**